src/Cake.Core/CakeTaskExtensions.cs

```diff
@@ -29,15 +46,24 @@ public static void AddDependee(this CakeTask task, string name, bool required =
             task.Dependees.Add(new CakeTaskDependency(name, required));
         }
 
-        public static void AddCriteria(this CakeTask task, Func<ICakeContext, bool> criteria, string message = null)
+        /// <summary>
+        /// Adds the criteria to the task's criterias.
+        /// </summary>
+        /// <param name="task">The task.</param>
+        /// <param name="predicate">The criteria's predicate.</param>
+        /// <param name="message">The criteria's message if skipped.</param>
+        public static void AddCriteria(this CakeTask task, Func<ICakeContext, bool> predicate, string message = null)
         {
-            if (criteria == null)
-            {
-                throw new ArgumentNullException(nameof(criteria));
-            }
-            task.Criterias.Add(new CakeTaskCriteria(criteria, message));
+            task.Criterias.Add(new CakeTaskCriteria(predicate, message));
```

Comment:
Note:
- I renamed the parameter from `criteria` to `predicate` to match the parameter name in the `CakeTaskCriteria` ctor
- I removed the check for null here because the `CakeTaskCriteria` ctor already does that